pycytominer/cyto_utils/cells.py

```diff
@@ -430,46 +435,210 @@ def aggregate_compartment(
+            conditional statements.
+
+            Parameters
+            ----------
+            df : pandas.core.frame.DataFrame
+                A dataframe with columns as strata and rows as unique value combos
+
+            dtypes : Dict[str, str]
+                Dictionary to look up SQLite datatype based on column name
+
+            n : int
+                Number of rows of the input df to combine in each conditional
+                statement
+
+            Returns
+            -------
+            grouped_conditions : List[str]
+                A list of strings, each string being a valid SQLite conditional
+
+            Example
+            -------
+
+                df:
+                    TableNumber | ImageNumber
+                    -------------------------
+                    [1]         | [1]
+                    [2]         | [1, 2, 3]
+                    [3]         | [1, 2]
+                    [4]         | [1]
+
+                _sqlite_strata_conditions(df,
+                                          dtypes={'TableNumber': 'integer', 'ImageNumber': 'integer'},
+                                          n=1):
+                    ['(TableNumber in (1) and ImageNumber in (1))',
+                     '(TableNumber in (2) and ImageNumber in (1, 2, 3))',
+                     '(TableNumber in (3) and ImageNumber in (1, 2))',
+                     '(TableNumber in (4) and ImageNumber in (1))']
+
+                _sqlite_strata_conditions(df,
+                                          dtypes={'TableNumber': 'text', 'ImageNumber': 'integer'},
+                                          n=2):
+                    ["(TableNumber in ('1') and ImageNumber in (1))
+                      or (TableNumber in ('2') and ImageNumber in (1, 2, 3))",
+                     "(TableNumber in ('3') and ImageNumber in (1, 2))
+                      or (TableNumber in ('4') and ImageNumber in (1))"]
+            """
+            conditions = []
+            for row in df.iterrows():
+                series = row[1]
+                values = [
+                    [f"'{a}'" for a in y] if dtypes[x] == "text" else y
+                    for x, y in zip(series.index, series.values)
+                ]  # put quotes around text entries
+                condition_list = [
+                    f"{x} in ({', '.join([str(a) for a in y]) if len(y) > 1 else y[0]})"
+                    for x, y in zip(series.index, values)
+                ]
+                conditions.append(f"({' and '.join(condition_list)})")
+            grouped_conditions = [
+                " or ".join(conditions[i : (i + n)])
+                for i in range(0, len(conditions), n)
+            ]
+            return grouped_conditions
+
+        # Translate the first strata combination into a SQLite condition string
+        first_stratum_condition = _sqlite_strata_conditions(
+            df=df_unique_mergecols.head(1),
```

Comment:
it took me a while to figure out that `df_unique_mergecols` is defined out of scope of this function. Should you include as a function argument in `_sqlite_strata_conditions`?

Reply:
`df_unique_mergecols` is in scope of `_compartment_df_generator()`, but out of scope of `_sqlite_strata_conditions()`